python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -574,18 +577,53 @@ def current_in_flight_tasks(self) -> int:
             for actor_state in self._running_actors.values()
         )
 
-    def scale_up(self, num_actors: int) -> int:
+    def can_scale_down(self):
+        # NOTE: To prevent bouncing back and forth, we disallow scale down for
+        #       a "cool-off" period after the most recent scaling up, with an intention
+        #       to allow application to actually utilize newly provisioned resources
+        #       before making decisions on subsequent action
+        #
+        #       Note that this action is unidirectional and doesn't apply to
+        #       scaling up, ie if actor pool just scaled down, it'd still be able
+        #       to scale back up immediately.
+        return (
+            self._last_scale_up_ts is None
+            or time.time()
+            >= self._last_scale_up_ts + self._ACTOR_POOL_SCALE_DOWN_DEBOUNCE_PERIOD_S
+        )
+
+    def scale_up(self, num_actors: int, *, reason: Optional[str] = None) -> int:
+        logger.info(
+            f"Scaling up actor pool by {num_actors} ("
```

Comment:
```suggestion
            f"Scaling up actor pool by {num_actors}, current actor counts: ("
```

Reply:
It also includes reason so that needs to be more generic